File: study-helper-app/app/src/main/java/com/zybooks/studyhelper/repo/QuestionDao.java
package com.zybooks.studyhelper.repo;

import androidx.room.*;
import com.zybooks.studyhelper.model.Question;
import java.util.List;
import androidx.lifecycle.LiveData;

@Dao
public interface QuestionDao {
    @Query("SELECT * FROM Question WHERE id = :id")
    LiveData<Question> getQuestion(long id);

    @Query("SELECT * FROM Question WHERE subject_id = :subjectId ORDER BY id")
    LiveData<List<Question>> getQuestions(long subjectId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long addQuestion(Question question);

    @Update
    void updateQuestion(Question question);

    @Delete
    void deleteQuestion(Question question);
}

File: app/src/main/java/com/cs407/memora/repo/QuestionDao.kt
package com.cs407.memora.repo

import androidx.room.*
import com.cs407.memora.model.Question
import androidx.lifecycle.LiveData

@Dao
interface QuestionDao {
    @Query("SELECT * FROM Question WHERE id = :id")
    fun getQuestion(id: Long): LiveData<Question?>

    @Query("SELECT * FROM Question WHERE subject_id = :subjectId ORDER BY id")
    fun getQuestions(subjectId: Long): LiveData<List<Question>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addQuestion(question: Question): Long

    @Update
    fun updateQuestion(question: Question)

    @Delete
    fun deleteQuestion(question: Question)
}